engine/src/main/java/org/hibernate/validator/parameternameprovider/ReflectionParameterNameProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.parameternameprovider;
+
+import static org.hibernate.validator.internal.util.CollectionHelper.newArrayList;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.List;
+
+import javax.validation.ParameterNameProvider;
+
+/**
```

Comment:
I'd suggest to remove the documentation apart from the deprecation note, making it even less attractive to use the type. It's a pattern sucessfully employed by other frameworks, too.

Reply:
Done.